File: internal/order/queue.go
package order

// InsertPending inserts o into pending by VIP-before-Normal rules.
// VIP: after all existing VIPs, before first Normal.
// Normal: append to tail.
func InsertPending(pending []*Order, o *Order) []*Order {
	if o.Type == TypeNormal {
		return append(pending, o)
	}
	idx := 0
	for idx < len(pending) && pending[idx].Type == TypeVIP {
		idx++
	}
	pending = append(pending, nil)
	copy(pending[idx+1:], pending[idx:])
	pending[idx] = o
	return pending
}

File: src/utils/queue.ts
import type { Order } from '@/store/types'

export function insertByPriority(pending: Order[], order: Order): void {
  const insertAt = pending.findIndex(o => {
    if (order.type === 'VIP' && o.type === 'NORMAL') return true
    if (order.type === 'NORMAL' && o.type === 'VIP') return false
    return o.id > order.id
  })
  pending.splice(insertAt === -1 ? pending.length : insertAt, 0, order)
}
